### PR TITLE
Hb logger

### PR DESCRIPTION
**Pull request by vremere-a**

### CODE REVIEW

File: sam:/Users/artem2/Mate_Academy/hibernate-cinema/logs/application.log [hunk 1]
```diff
@@ -0,0 +1,12 @@
+2020-10-15 16:28:37 INFO  Version:44 - HHH000412: Hibernate ORM core version 5.4.21.Final
```
Comment (fyarkiy) [inline sam:/Users/artem2/Mate_Academy/hibernate-cinema/logs/application.log]: How this log get into commit? can you delete it and commit it again?

File: src/main/java/com/dev/cinema/Main.java [hunk 2]
```diff
@@ -64,51 +68,54 @@ public static void main(String[] args) throws AuthenticationException {
...
+        shoppingCartService.addSession(ironManSession, user3);
+        shoppingCartService.addSession(spiderManSession, user3);
+        log.info("show Shopping cart by user # 3 " + shoppingCartService.getByUser(user3));
 
-        shoppingCartService.addSession(flashSession,user2);
-        shoppingCartService.addSession(spiderManSession,user2);
-        System.out.println(shoppingCartService.getByUser(user2));
+        shoppingCartService.addSession(flashSession, user2);
+        shoppingCartService.addSession(spiderManSession, user2);
+        log.info("show Shopping cart by user # 3 " + shoppingCartService.getByUser(user2));
```
Comment (fyarkiy) [inline src/main/java/com/dev/cinema/Main.java]: a lot of cool "info" but never used "error" and "warn"

**Review by fyarkiy — APPROVED** [review]

File: src/main/java/com/dev/cinema/Main.java (hunk 2 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/Main.java]: agree. make sense

File: sam:/Users/artem2/Mate_Academy/hibernate-cinema/logs/application.log (hunk 1 above)
Comment (KseniiaMakarova) [inline sam:/Users/artem2/Mate_Academy/hibernate-cinema/logs/application.log]: this file should not be present in this PR
watch the video tutorial about how to remove redundant files from `master` and apply the same procedure to this branch

File: src/main/resources/log4j.properties [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Root logger option
+log4j.rootLogger=INFO, file
```
Comment (KseniiaMakarova) [inline src/main/resources/log4j.properties]:
```suggestion
log4j.rootLogger=INFO, stdout, file
```

File: src/main/java/com/dev/cinema/Main.java [hunk 4]
```diff
@@ -64,51 +68,54 @@ public static void main(String[] args) throws AuthenticationException {
...
         user1.setPassword("123");
         userService.add(user1);
-        System.out.println(userService.findByEmail(user1.getEmail()));
+        log.info("Find User by email " + userService.findByEmail(user1.getEmail()));
 
         AuthenticationService authenticationService =
                 (AuthenticationService) injector.getInstance(AuthenticationService.class);
         authenticationService.register("arts2@ukr.net", "1234");
-        System.out.println(authenticationService.login("arts2@ukr.net", "1234"));
+        log.info(authenticationService.login("arts2@ukr.net", "1234"));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: let's wrap this in try-catch and add log.warn with the informative message in `catch` block

File: src/main/java/com/dev/cinema/Main.java [hunk 5]
```diff
@@ -64,51 +68,54 @@ public static void main(String[] args) throws AuthenticationException {
...
 
         User user2 = userService.findByEmail("arts2@ukr.net").get();
 
         authenticationService.register("arts3@ukr.net", "12345");
-        System.out.println(authenticationService.login("arts3@ukr.net", "12345"));
+        log.info(authenticationService.login("arts3@ukr.net", "12345"));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: same

File: src/main/java/com/dev/cinema/Main.java (hunk 2 above)
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]:
```suggestion
        log.info("show Shopping cart by user # 2 " + shoppingCartService.getByUser(user2));
```

File: src/main/java/com/dev/cinema/Main.java [hunk 6]
```diff
@@ -64,51 +68,54 @@ public static void main(String[] args) throws AuthenticationException {
...
+        log.info("show Shopping cart by user # 3 " + shoppingCartService.getByUser(user2));
         shoppingCartService.clear(shoppingCartService.getByUser(user2));
 
         OrderService orderService =
                 (OrderService) injector.getInstance(OrderService.class);
         orderService.completeOrder(shoppingCartService.getByUser(user3).getTickets(), user3);
-        orderService.getOrderHistory(user3).forEach(System.out::println);
+        log.info("getting all order history");
+        orderService.getOrderHistory(user3).forEach(log::info);
+        log.info("APPLICATION EXITS");
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java:127]: it will be nice to have logs from DaoImpl classes

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/resources/log4j.properties (hunk 3 above)
Comment (vremere-a) [reply src/main/resources/log4j.properties]: done

File: src/main/java/com/dev/cinema/Main.java (hunk 4 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/Main.java]: done
try {
            authenticationService.login("arts2@ukr.net", "1234");
        } catch (AuthenticationException e) {
            log.info("Incorrect username or password");
        }

File: src/main/java/com/dev/cinema/Main.java (hunk 5 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/Main.java]: done

File: src/main/java/com/dev/cinema/Main.java (hunk 2 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/Main.java]: done

File: src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 7]
```diff
@@ -25,6 +27,7 @@ public ShoppingCart add(ShoppingCart shoppingCart) {
             transaction = session.beginTransaction();
             session.save(shoppingCart);
             transaction.commit();
+            log.info("Entity " + shoppingCart + "successful added from the DB");
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java:30]:
```suggestion
            log.info("Entity " + shoppingCart + "successful added to the DB");
```
everywhere
and what about `update()` method?

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**